pkg/gofr/datasource/solr/solr.go

```diff
@@ -35,8 +35,8 @@ type Client struct {
 // client := New(config)
 // client.UseLogger(loggerInstance)
 // client.UseMetrics(metricsInstance)
-// client.Connect()
-func New(conf Config) *Client {
+// client.Connect().
+func New(conf *Config) *Client {
```

Comment:
This change is unnecessary and does not feel like a good idea. Config should not be a pointer.

Reply:
I agree.

https://github.com/gofr-dev/gofr/pull/1245#discussion_r1860112867

Reply:
I was trying to be consistent across all the datasources. Some datasources that have a heavy config struct are passed by reference. So was being consistent across datasources passing each config struct by reference.

Reply:
The same point has been raised by myself, on another PR

https://github.com/gofr-dev/gofr/pull/1254#discussion_r1862569427

Reply:
I don't know what should be done. I agree it's better to have a consistent say to do.

But having a pointer for a configuration looks strange, not, yes it has performance/memory issues

Reply:
Would be changing it to a pointer to struct.